File: src/Vult.WebApp/projects/vult/src/app/core/models/product.ts
// Copyright (c) Quinntyne Brown. All Rights Reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

export interface Product {
  productId: string;
  estimatedMSRP?: number;
  estimatedResaleValue?: number;
  description?: string;
  size?: string;
  brandName?: string;
  gender?: Gender;
  itemType?: ItemType;
  isFeatured?: boolean;
  createdDate: string;
  updatedDate: string;
  productImages?: ProductImage[];
}

export interface ProductImage {
  productImageId: string;
  productId: string;
  imageData?: string;
  url?: string;
  description?: string;
  createdDate: string;
}

export interface IngestionResult {
  success: boolean;
  errors: string[];
  totalProcessed: number;
  successfullyProcessed: number;
  failed: number;
  products: Product[];
}

export interface IngestionProgress {
  current: number;
  total: number;
  status: string;
}

export enum Gender {
  Mens = 0,
  Womens = 1,
  Unisex = 2
}

export enum ItemType {
  Shoe = 0,
  Pants = 1,
  Jacket = 2,
  Shirt = 3,
  Shorts = 4,
  Dress = 5,
  Skirt = 6,
  Sweater = 7,
  Hoodie = 8,
  Coat = 9,
  Bag = 10,
  Accessories = 11,
  Hat = 12,
  Book = 13
}

File: src/Vult.Core/Model/ProductAggregate/Product.cs
// Copyright (c) Quinntyne Brown. All Rights Reserved.
// Licensed under the MIT License. See LICENSE in the project root for license information.

using Vult.Core.Model.ProductAggregate.Enums;

namespace Vult.Core.Model.ProductAggregate;

public class Product
{
    public Guid ProductId { get; set; }

    public decimal EstimatedMSRP { get; set; }

    public decimal EstimatedResaleValue { get; set; }

    public string Description { get; set; } = string.Empty;

    public string ShortDescription { get; set; } = string.Empty;

    public string Size { get; set; } = string.Empty;

    public string BrandName { get; set; } = string.Empty;

    public Gender Gender { get; set; }

    public ItemType ItemType { get; set; }

    public bool IsFeatured { get; set; }

    public string Benefits { get; set; } = string.Empty;

    public string Details { get; set; } = string.Empty;

    public string Shipping { get; set; } = string.Empty;

    public string PromotionalMessage { get; set; } = string.Empty;

    public bool IsMemberExclusive { get; set; }

    public bool IsNew { get; set; }

    public DateTime CreatedDate { get; set; }

    public DateTime UpdatedDate { get; set; }

    // Navigation property
    public ICollection<ProductImage> ProductImages { get; set; } = new List<ProductImage>();
}
